### PR TITLE
[Pybind] update to 2.4.3

### PR DESCRIPTION
**Pull request by philbucher**
https://github.com/pybind/pybind11/releases/tag/v2.4.3

so far we use 2.2.4: #2877

### CODE REVIEW

**Comment by roigcarlo** [comment]
ping @KratosMultiphysics/technical-committee

**Comment by philbucher** [comment]
I will also try this with the intel compiler

**Comment by philbucher** [comment]
ping @pooyan-dadvand @KratosMultiphysics/altair please try this on your machines

**Comment by philbucher** [comment]
I checked this with the Intel compiler version `19.0.5.20190815` on [SuperMUC-NG @ LRZ](https://doku.lrz.de/display/PUBLIC/SuperMUC-NG)
I ran the tests for the Core and a few Apps

Build Type | OpenMP configuration | Tests results
--- | --- | ---
Release | 1 threads | All good
Release | 2 threads | All good
Release | 3 threads | All good
Release | 4 threads | All good
Release | disabled | All good
FullDebug | 1 threads | All good
FullDebug | 2 threads | All good
FullDebug | 3 threads | All good
FullDebug | 4 threads | All good
FullDebug | disabled | All good

Good news is that also the problems with the intel compiler seem to have disappeared, see #2690
This was already working on Master too

**TL;DR**
Works, lets merge

**Comment by pooyan-dadvand** [comment]
Checked with gcc 4.8 in centos7 and also works fine.

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee approves this
